cmd/pd-server/main.go

```diff
@@ -224,7 +224,31 @@ func start(cmd *cobra.Command, args []string, services ...string) {
 
 	var sig os.Signal
 	go func() {
-		sig = <-sc
+	SignalHandler:
+		for {
+			select {
+			case sig = <-sc:
+				log.Info("Received a signal", zap.String("signal", sig.String()))
+				if svr.GetMember().IsLeader() {
+					log.Info("Transferring leadership due to signal", zap.String("signal", sig.String()))
+					err := svr.GetMember().ResignEtcdLeader(context.TODO(), svr.Name(), "")
+					if err != nil {
```

Comment:
maybe you needs seconds to wait the pd leader has transfer also?

Reply:
Thanks for the suggestion @bufferflies , I have updated the PR. Are there any other ways of resigning (including waiting until another leader has taken over)?

Reply:
PD should grant new lease after the new pd leader compaign success ,so you can use watch api to listen the change .